operator/src/test/java/oracle/kubernetes/operator/MainTest.java

```diff
@@ -196,6 +206,62 @@ private V1ObjectMeta createMetadata(DateTime creationTimestamp) {
         .resourceVersion("1");
   }
 
+  @Test
+  public void whenConfiguredDomainNamespaceMissing_logWarning() {
+    loggerControl.withLogLevel(Level.WARNING).collectLogMessages(logRecords, MessageKeys.NAMESPACE_IS_MISSING);
+
+    defineSelectionStrategy(Main.Namespaces.SelectionStrategy.List);
+    String namespaceString = "NS1,NS" + LAST_NAMESPACE_NUM;
+    HelmAccessStub.defineVariable(HelmAccess.OPERATOR_DOMAIN_NAMESPACES, namespaceString);
+    createNamespaces(LAST_NAMESPACE_NUM - 1);
+
+    testSupport.runSteps(new Main.Namespaces(false).readExistingNamespaces());
+
+    assertThat(logRecords, containsWarning(MessageKeys.NAMESPACE_IS_MISSING));
+  }
+
+  @Test
+  public void whenNamespacesListedInOneChunk_dontDeclarePresentNamespacesAsMissing() {
+    loggerControl.withLogLevel(Level.WARNING).collectLogMessages(logRecords, MessageKeys.NAMESPACE_IS_MISSING);
+
+    defineSelectionStrategy(Main.Namespaces.SelectionStrategy.List);
+    String namespaceString = "NS1,NS" + LAST_NAMESPACE_NUM;
+    HelmAccessStub.defineVariable(HelmAccess.OPERATOR_DOMAIN_NAMESPACES, namespaceString);
+    createNamespaces(LAST_NAMESPACE_NUM);
+
+    testSupport.runSteps(new Main.Namespaces(false).readExistingNamespaces());
+  }
+
+  // todo add unit tests for namespace shutdown. Must manipulate the namespaceStopping map. Can that be abstracted?
+
+  private void defineSelectionStrategy(SelectionStrategy selectionStrategy) {
+    TuningParameters.getInstance().put(Main.Namespaces.SELECTION_STRATEGY_KEY, selectionStrategy.toString());
+  }
+
+  @Test
+  public void whenNamespacesListedInMultipleChunks_dontDeclarePresentNamespacesAsMissing() {
```

Comment:
what is the difference between this test and the whenNamespacesListedInOneChunk...()?

Reply:
The difference is the number of namespaces. A chunk, by default, is 50 items (see the following code in TuningParametersImpl.java:```
    CallBuilderTuning callBuilder =
        new CallBuilderTuning(
            *(int) readTuningParameter("callRequestLimit", 50)*,
            (int) readTuningParameter("callMaxRetryCount", 5),
            (int) readTuningParameter("callTimeoutSeconds", 10));
```

In the old code, the detection of whether a namespace was stranded was being done on each chunk, meaning that any namespace not in the current chunk was incorrectly reported as stranded. The new code does the check after all namespaces have been located.